src/HttpWorker.php

```diff
@@ -154,6 +172,37 @@ private function createRequest(string $body, array $context): Request
         );
     }
 
+    private function requestFromProto(string $body, RequestProto $message): Request
+    {
+        /** @var UploadedFilesList $uploads */
+        $uploads = \json_decode($message->getUploads(), true) ?? [];
+        $headers = $this->headerValueToArray($message->getHeader());
+
+        \parse_str($message->getRawQuery(), $query);
+        /** @psalm-suppress ArgumentTypeCoercion, MixedArgumentTypeCoercion */
+        return new Request(
+            remoteAddr: $message->getRemoteAddr(),
+            protocol: $message->getProtocol(),
+            method: $message->getMethod(),
+            uri: $message->getUri(),
+            headers: $this->filterHeaders($headers),
+            cookies: \array_map(
+                static fn(array $values) => \implode(',', $values),
+                $this->headerValueToArray($message->getCookies()),
+            ),
+            uploads: $uploads,
+            attributes: [
+                Request::PARSED_BODY_ATTRIBUTE_NAME => $message->getParsed(),
+            ] + \array_map(
+                static fn(array $values) => \array_shift($values),
+                $this->headerValueToArray($message->getAttributes()),
+            ),
+            query: $query,
+            body: $message->getParsed() && empty($body) ? \json_encode([]) : $body,
```

Comment:
`\json_encode([])` can be replaced with a scalar.
BTW can't it be an empty string?